docs: Move include markdown files not used by help center.

laurynmm opened the pull request:
Moves files in `templates/zerver/help/include` that are not used by the help center documentation to a new directory, `templates/zerver/templates_include`.

Updates the `base_path` used for the making the custom markdown include extension in `render_markdown_path` to be based on whether the documentation page is part of the help center or not.

**Notes & Testing**:
- Prep commit for work on updating the help center directory structure. See [relevant CZO conversation](https://chat.zulip.org/#narrow/stream/19-documentation/topic/help.20center.20directory.20structure/near/1436396).
- Checking via git-grep, there are no include files currently used in the corporate pages, and therefore policy doc pages as well.
- Ran a diff against the current main branch with these changes for both the help center and API documentation and saw no changes in the rendered HTML. Did manual testing of the integrations documentation.

**Questions**:
- I wasn't sure about the naming for the directory for non-help center include files. I went with `templates_include` for this first draft to get feedback.
  - There is already `templates/zerver/include`, which is used for a template specific to the desktop app, `back_to_login_component.html`.
  - One option would be to move that file to another directory,  and use this directory for the include macro markdown files?
  - We could also create a sub-directory, `templates/zerver/include/md_macro/`?
- There were 4 files in `templates/zerver/help/include` that do not seem to be in use that we may want to delete?
  - ` cloud-plus-only.md`
  - `go-to-stream-via-all-streams.md`
  - `running-a-bot.md` *note: I moved this to the new directory in this draft*
  - `subscriptions.md`

---

**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.

alya commented:
> cloud-plus-only.md

This one is used on /help/saml-authentication.

alya commented:
> go-to-stream-via-all-streams.md

Can you check where it was used before / when we stopped using it, so that we can think briefly about whether it's useful?

alya commented:
> subscriptions.md

This is one line; I'm happy to drop it without investigation if we aren't using it.

laurynmm commented:
**Updates re: unused files**:
- It looks like the intention was to use `cloud-plus-only.md` as a macro on `saml_authentication`, but it wasn't. I've make that update so that the file is used.
- It looks like `go-to-stream-via-all-streams` was removed in 1318cdfcc. I've deleted it in a prep commit here.
- It looks like `running-a-bot` never got used as far as I can tell. It was introduced in 115328868 and the commit message implies that there was an expectation to use it. I've removed it in a prep commit here.
- I've removed `subscriptions.md` in a prep commit as well.

---

The naming question still stands. I find that I'm leaning towards moving `back_to_login_component.html` up to the `templates/zerver` directory, and then using the `templates/zerver/include` directory for the markdown macro files that are not help center files.

alya commented:
> * It looks like the intention was to use `cloud-plus-only.md` as a macro on `saml_authentication`, but it wasn't. I've make that update so that the file is used.

I'm super confused -- I do see it in that file.

```
If you are [self-hosting](/self-hosting/) Zulip, please follow the detailed setup instructions in
the [SAML configuration for self-hosting][saml-readthedocs]. The documentation
on this page may be a useful reference for how to set up specific SAML
providers.

{!cloud-plus-only.md!}
```

alya commented:
All the rest makes sense to me!

laurynmm commented:
> > * It looks like the intention was to use `cloud-plus-only.md` as a macro on `saml_authentication`, but it wasn't. I've make that update so that the file is used.
> 
> I'm super confused -- I do see it in that file.

If you look at `saml_authentication.md` on main, https://raw.githubusercontent.com/zulip/zulip/main/templates/zerver/help/saml-authentication.md, and search for the macro, it's not there.

I added a prep commit here that switches the warning text in `saml_authentication.md` with the `cloud-plus-only` macro. Looks like it was just missed in the commit that added the file: 95bc77ae4f.

timabbott commented:
I realized that our Python API repository does define some files that can use these macros (it gets copied into `static/generated/bots` in zulip/zulip, I think? See `tools/setup/generate_zulip_bots_static_files.py`). 

`python-zulip-api$ find . -name \*.md`  works to find all that might do this (I think only the `zulip_bots` directory does this currently?).

Grepping, I think none of them actually use the removed macros, so we should be OK.

timabbott commented on zerver/lib/templates.py:
```diff
@@ -146,6 +145,14 @@ def render_markdown_path(
+        )
+    else:
+        md_macro_extension = zerver.lib.markdown.include.makeExtension(
+            base_path="templates/zerver/include/markdown_macros/"
```
What's the thinking behind `include/markdown_macros` vs say `include/api` and `include/integrations`, or `templates/zerver/markdown_macros/`?

timabbott replied:
Might be a good #documentation discussion as it's kinda a question of "what feels intuitive to others".

timabbott replied:
I wonder if we could make this just read the `api/` part from the path without doing something hideous, so that `templates/zerver/api/` can use `templates/zerver/include/api` without a chain of if/else statements. I suppose the simplest would be passing a `include_namespace="help"` parameter rather than `help_center=True`.

We can always share macros between /api and /integrations via a symlink if there's one that's used in both places.

laurynmm replied:
> I wonder if we could make this just read the `api/` part from the path without doing something hideous, so that `templates/zerver/api/` can use `templates/zerver/include/api` without a chain of if/else statements. I suppose the simplest would be passing a `include_namespace="help"` parameter rather than `help_center=True`.

I did think about that, but since this is a prep for moving the help center documentation out of `templates/zerver`, I thought that the boolean was a better option. The follow-up to this will be moving those help center files and the base path for the include would then be completely different.

> What's the thinking behind `include/markdown_macros` vs say `include/api` and `include/integrations`, or `templates/zerver/markdown_macros/`?
>
> Might be a good #documentation discussion as it's kinda a question of "what feels intuitive to others".

I can definitely start that conversation. I can see `templates/zerver/markdown_macros` being a more intuitive option that what I used.

timabbott replied:
Ahh, yeah, I'd forgotten that we were planning to move it further out of `templates/zerver`. One thing on my mind is whether we want something like:

`templates/help/` for the /help/: and `templates/include/help/` for its include content.

That path would kinda signal that we'd have a "help" Django application in Django's file hierarchy; which is probably not a bad impression to signal even if we're not actually doing the bit of overhead to make one.

laurynmm commented:
**Updates**:
- Split moving the impacted files into two commits so that the integrations and API documentation include markdown macros are in different directories.
  - The 3rd commit:
    - Moves the integrations documentation files to `templates/zerver/integrations/include`. 
    - Creates a boolean parameter in `render_markdown_path` for the integrations docs.
  - The 4th commit:
    - Moves the API documentation files to `templates/zerver/api/include`.
    - Also moves the test file, `empty.md`.
    - Creates a boolean parameter in `render_markdown_path` for the help center.
- I went with adding two boolean parameters instead of a string that could be used to make the intended file path for a couple of reasons:
  - Even though these directory paths are all very similar (e.g. `templates/zerver/XXX/include`) in these changes, moving the separate types of documentation will be easier in follow-up tasks if we don't have to keep the directory path the same.
  - If we do end up with similar directory paths after restructuring the different types of documentation, it would be easy to do a follow-up where we change the booleans to a string and remove the if/else blocks.
- The API documentation directory is the default in this implementation. That could be adjusted so that it's the integrations or help center docs. Or we could also have a boolean parameter for the API docs and have a default directory for shared files.
  - Currently there are no include markdown macro files that are shared between documentation types. The tutorial for incoming webhooks does include examples of the macro formatting, but those shouldn't be rendered as include macros.

timabbott commented:
> Even though these directory paths are all very similar (e.g. templates/zerver/XXX/include) in these changes, moving the separate types of documentation will be easier in follow-up tasks if we don't have to keep the directory path the same.

I don't have strong feelings on this -- one can always do conditionals on the values, but I'm happy to defer to the person doing the work on this sort of thing.

timabbott commented:
This looks good, merged, thanks @laurynmm!

One consequence of moving these URLs is that one can now try accessing `/api/include/empty`, for example. I didn't find a clear bug accessing any of those (the /integrations behavior is a bit weird -- you get the /integrations homepage if you pass a random URL -- but that seems not a problem related to this), but figured I'd flag this as something to think about; we do have different configuration here. (For example, `/api/foo` I think doesn't accept anything with a `/` in it, but `/help/` does).

```
help/<path:article>
api/<slug:article>
```
